plugins/woocommerce/includes/wc-conditional-functions.php

```diff
@@ -136,6 +136,20 @@ function is_checkout_pay_page() {
 	}
 }
 
+if ( ! function_exists( 'is_pay_for_order_page' ) ) {
+
+	/**
+	 * is_pay_for_order_page - Returns true when viewing the pay for order page.
+	 *
+	 * @param bool $check_key Optional. If true, check if the key is set in the URL.
+	 * @return bool
+	 */
+	function is_pay_for_order_page( $check_key = false ) {
```

Comment:
Since we introduce two new functions, let's ensure those are strictly typed (parameters, return types - as per the support language level).